R/teal_data-get_code.R

```diff
@@ -108,6 +108,7 @@ setMethod("get_code", signature = "teal_data", definition = function(object, dep
   checkmate::assert_flag(deparse)
 
   code <- if (!is.null(datanames)) {
+    warning("get_code(datanames) was deprecated in teal.data 0.6.1, use get_code(names) instead.", call. = FALSE)
```

Comment:
I was unlucky with using

```r
lifecycle::deprecate_soft(
  when = " 0.6.1", 
  what = "get_code(datanames)", 
  with = "get_code(names)", 
  always = TRUE
)
```

Reply:
What was the error?

Reply:
I think I had issues with `deprecate_soft`, but with `deprecate_warn` this is working

```r
> devtools::load_all(".")
ℹ Loading teal.data
Loading required package: teal.code
> library(teal.data)
> tdata1 <- within(teal_data(), {a <- 1; b <- 2})
> get_code(tdata1)
[1] "a <- 1\nb <- 2"
> get_code(tdata1, datanames = "a")
[1] "a <- 1"
Warning message:
The `datanames` argument of `get_code()` is deprecated as of teal.data
0.6.1.
ℹ Please use the `names` argument of `teal.code::get_code()` instead.
ℹ The deprecated feature was likely used in the teal.data package.
  Please report the issue at
  <https://github.com/insightsengineering/teal.data/issues>.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated. 
```

Reply:
Pushed the changes using lifecycle package https://github.com/insightsengineering/teal.data/pull/343/commits/3942fecb9027fc6689cae15e6e0d02aa113951d8